test/tools/router_check/test/route_tests.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+set -e
+
+# Router_check_tool binary path
+PATH_BIN="test/tools/router_check/router_check_tool"
+
+# Config json path
+PATH_CONFIG="test/tools/router_check/test/config"
+
+TESTS=("ContentType" "ClusterHeader" "HeaderMatchedRouting" "Redirect" "Redirect2" "Redirect3" "TestRoutes" "Weighted")
+
+# Testing expected matches
+for t in "${TESTS[@]}"
+do
+  TEST_OUTPUT=$("${PATH_BIN}" "${PATH_CONFIG}/${t}.json" "${PATH_CONFIG}/${t}.golden.json" "--details")
+done
+
+# Bad config file
+BAD_CONFIG_OUTPUT=$(("${PATH_BIN}" "${PATH_CONFIG}/Redirect.golden.json" "${PATH_CONFIG}/TestRoutes.json") 2>&1) ||
+  if [[ "${BAD_CONFIG_OUTPUT}" == *"config schema JSON load failed"* ]]; then
+    echo testing bad config output
+    echo ${BAD_CONFIG_OUTPUT}
+  else
+    exit 1
+  fi
+
+# Failure test case
+FAILURE_OUTPUT=$("${PATH_BIN}" "${PATH_CONFIG}/TestRoutes.json" "${PATH_CONFIG}/Weighted.golden.json" "--details") ||
+  if [[ "${FAILURE_OUTPUT}" == *"cluster1 instant-server cluster_name"* ]]; then
```

Comment:
In this test, I mixed and matched the TestRoutes router config with the Weighted test tool config. In that case there will be failure test cases. Rather than matching on all the failure test cases, I am checking that a specific cluster_name match test case failed. The failed test case has cluster1 as the expected cluster name and instant-server as the actual cluster name returned by the router.